worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala

```diff
@@ -76,14 +76,15 @@ abstract private[worker] class Flusher(
                     flushTimeMetric.update(delta)
                   }
                 } catch {
-                  case _: ClosedByInterruptException =>
                   case e: IOException =>
                     task.notifier.setException(e)
                     processIOException(e, DiskStatus.READ_OR_WRITE_FAILURE)
+                  case t =>
+                    logWarning(s"Flusher-${this}-thread-${index} encounter exception.", t)
                 }
                 lastBeginFlushTime.set(index, -1)
               }
-              returnBuffer(task.buffer)
+              Utils.tryLogNonFatalError(returnBuffer(task.buffer))
```

Comment:
in which cases, `returnBuffer` may throw exceptions? I saw other two places invoking it without handle any exceptions

Reply:
If a netty's buffer reference count is incorrect. This method will throw illegalreferencecountexception.

Reply:
please add comment here